File: scratch/fix_encoding.js

const fs = require('fs');
const path = 'c:\\PROYECTOS\\DIICZONE_APP\\components\\workstation\\cm\\CMWorkstationLayout.js';

try {
    let data = fs.readFileSync(path);
    let text = data.toString('utf8');

    // Fixing mangled quotes and other characters
    let fixed = text
        .replace(/\\>/g, '">')
        .replace(/\\ /g, '" ')
        .replace(/\\"/g, '"')
        .replace(/\\}/g, '"}')
        .replace(/\\\)/g, '")')
        .replace(/\\\[/g, '"[')
        .replace(/\\\]/g, '"]')
        .replace(/\\</g, '"<')
        .replace(/\\n/g, '\n')
        .replace(/Sincronizaci\?n/g, 'Sincronización')
        .replace(/\?LITE/g, 'ÉLITE')
        .replace(/Configuraci\?n/g, 'Configuración')
        .replace(/Pr\?ximamente/g, 'Próximamente');

    // Also fix the case where backslash was used instead of double quote for attribute values
    // e.g. className=\bg-gradient... -> className="bg-gradient..."
    fixed = fixed.replace(/className=\\([^\s>]+)/g, 'className="$1"');
    
    // More aggressive quote fixing
    fixed = fixed.replace(/=\\([^\s>]+)/g, '="$1"');

    fs.writeFileSync(path, fixed, 'utf8');
    console.log('File recovery attempt 1 complete.');
} catch (err) {
    console.error('Error fixing file:', err);
}

File: scratch/fix_encoding.py

import os

path = r'c:\PROYECTOS\DIICZONE_APP\components\workstation\cm\CMWorkstationLayout.js'

with open(path, 'rb') as f:
    data = f.read()

# Try to find where the mangled part starts.
# It likely starts where I added the CMProfileView or where the first backslash-quote appeared.
# The error said index 177749.

# We'll try to fix common mangling:
# 1. Replace \> with ">
# 2. Replace \ with " (if it looks like a quote)
# 3. Fix the ? characters if possible, or just replace them with the correct ones.

try:
    text = data.decode('utf-8', errors='ignore')
except:
    text = data.decode('latin-1', errors='ignore')

# Fixing the mangled quotes
fixed_text = text.replace('\\>', '">').replace('\\ ', '" ').replace('\\"', '"').replace('\\}', '"}').replace('\\)', '")').replace('\\[', '"[').replace('\\]', '"]').replace('\\<', '"<')

# The user reported "R" and "?" characters.
# Sincronizaci?n -> Sincronización
# ?LITE -> ÉLITE
fixed_text = fixed_text.replace('Sincronizaci?n', 'Sincronización').replace('?LITE', 'ÉLITE').replace('Configuraci?n', 'Configuración').replace('Pr?ximamente', 'Próximamente')

with open(path, 'w', encoding='utf-8') as f:
    f.write(fixed_text)

print("File fixed attempted.")
